### PR TITLE
Update gke deployment scripts

### PR DESCRIPTION
**Pull request by juliawu**
As a followup to https://github.com/datacommonsorg/website/pull/3781, this PR:
* Completes the TODO of adding API Key generation and secret creation to the `create_api_key.sh script.`
* Adds the missing step of granting logging and monitoring permissions to the default compute engine service account into the `create_cluster.sh` script, which is needed for logging
* Adds the missing step of running `deploy_gke_helm.sh` at the end of one-time setup.

### CODE REVIEW

File: gke/README.md [hunk 1]
```diff
@@ -62,29 +62,17 @@ the file to describe the environment the clusters are being used for.
...
-
-3. Run the following scripts sequentially.
+1. Run the following scripts sequentially.
   
 ```bash
 # Create clusters
 ./create_all_clusters.sh <ENV>
 
 # Set up multi-cluster ingress and service
 ./setup_config_cluster.sh
```
Comment (shifucun) [inline gke/README.md:72]: Move ./setup_config_cluster.sh to the end as optional step?

File: gke/create_api_key.sh [hunk 2]
```diff
@@ -34,4 +34,32 @@ KEY_STRING=$(gcloud alpha services api-keys get-key-string $API_KEY_NAME --forma
 touch /tmp/dc-website-api-key
 > /tmp/dc-website-api-key
 echo "$KEY_STRING" >> /tmp/dc-website-api-key
-gcloud secrets create maps-api-key --data-file=/tmp/dc-website-api-key
\ No newline at end of file
+gcloud secrets create maps-api-key --data-file=/tmp/dc-website-api-key
+
+# Create API key for Data Commons API
```
Comment (shifucun) [inline gke/create_api_key.sh:39]: this is very nice!

File: gke/create_api_key.sh [hunk 3]
```diff
@@ -34,4 +34,32 @@ KEY_STRING=$(gcloud alpha services api-keys get-key-string $API_KEY_NAME --forma
...
+  --allowed-referrers=$DOMAIN \
+  --api-target=service=generativelanugage.googleapis.com
+
+API_KEY_NAME=$(gcloud alpha services api-keys list --filter='displayName=nl-palm-api-key' --format='value(name)')
+KEY_STRING=$(gcloud alpha services api-keys get-key-string $API_KEY_NAME --format='value(keyString)')
+
+touch /tmp/palm-api-key
+> /tmp/palm-api-key
+echo "$KEY_STRING" >> /tmp/palm-api-key
+gcloud secrets create palm-api-key --data-file=/tmp/palm-api-key
```
Comment (shifucun) [inline gke/create_api_key.sh:65]: nit: add newline at the end of both .sh file

**Review by shifucun — APPROVED** [review]
Thanks for updating the doc and figuring out the logging issue!

File: gke/README.md (hunk 1 above)
Comment (juliawu) [reply gke/README.md]: Done!

File: gke/create_api_key.sh (hunk 3 above)
Comment (juliawu) [reply gke/create_api_key.sh]: Done!

**Pull request merged**